clients/include/hipBuffer.hpp

```diff
@@ -0,0 +1,313 @@
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#pragma once
+
+#include <hip/hip_runtime.h>
+#include "hipblaslt_ostream.hpp"
+
+inline hipDataType computeTypeToRealDataType(hipblasComputeType_t ctype)
+{
+    static const std::map<hipblasComputeType_t, hipDataType> ctypeMap{{HIPBLAS_COMPUTE_16F, HIP_R_16F},
+                                                                    {HIPBLAS_COMPUTE_16F_PEDANTIC, HIP_R_16F},
+                                                                    {HIPBLAS_COMPUTE_32F, HIP_R_32F},
+                                                                    {HIPBLAS_COMPUTE_32F_PEDANTIC, HIP_R_32F},
+                                                                    {HIPBLAS_COMPUTE_32F_FAST_16F, HIP_R_32F},
+                                                                    {HIPBLAS_COMPUTE_32F_FAST_16BF, HIP_R_32F},
+                                                                    {HIPBLAS_COMPUTE_32F_FAST_TF32, HIP_R_32F},
+                                                                    {HIPBLAS_COMPUTE_64F, HIP_R_64F},
+                                                                    {HIPBLAS_COMPUTE_64F_PEDANTIC, HIP_R_64F},
+                                                                    {HIPBLAS_COMPUTE_32I, HIP_R_32I},
+                                                                    {HIPBLAS_COMPUTE_32I_PEDANTIC, HIP_R_32I}};
+
+    return ctypeMap.at(ctype);
+}
+
+inline std::size_t realDataTypeSize(hipDataType dtype)
+{
+    static const std::map<hipDataType, std::size_t> dtypeMap{{HIP_R_32F, 4},
+                                                             {HIP_R_64F, 8},
+                                                             {HIP_R_16F, 2},
+                                                             {HIP_R_8I, 1},
+                                                             {HIP_R_8U, 1},
+                                                             {HIP_R_32I, 4},
+                                                             {HIP_R_32U, 4},
+                                                             {HIP_R_16BF, 2},
+                                                             {HIP_R_4I, 1},
+                                                             {HIP_R_4U, 1},
+                                                             {HIP_R_16I, 2},
+                                                             {HIP_R_16U, 2},
+                                                             {HIP_R_64I, 8},
+                                                             {HIP_R_64U, 8},
+                                                             {HIP_R_8F_E4M3_FNUZ, 1},
+                                                             {HIP_R_8F_E5M2_FNUZ, 1}};
+
+    return dtypeMap.at(dtype);
+}
+
+union computeTypeInterface{
+    float f32;
+    double f64;
+    hipblasLtHalf f16;
+    int32_t i32;
+};
```

Comment:
Please consider how to apply d_vector into hipDeviceBuffer. There are some MEMORY guard testing in d_vector.

Reply:
Completed. It works, I did a small test of polluted and non-polluted values for hipDeviceBuffer.